### PR TITLE
variable name change for better understanding

### PR DESCRIPTION
**Pull request by kushagra98**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
TL;DR: Changed variable name from `targetUtilization`to `targetUsage` in the fuction [calcPlainMetricReplicas](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/podautoscaler/replica_calculator.go#L178) for better understanding of the code.

Here, these metrics are non-utlilization (non-percentage) metrics. These metrics here are the usage metrics like 10m CPU, 1G memory, etc. The variable `targetUtilization` here seems to be a bad copy paste from above function [GetResourceReplicas](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/podautoscaler/replica_calculator.go#L65) is now changed to `targetUsage` for better understanding of the code.

#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by kushagra98** [comment]
Attaching discussion with @mwielgus from the PR #112178
image

**Comment by kushagra98** [comment]
@pbeschetnov @pbetkier can you look at this change?

File: pkg/controller/podautoscaler/replica_calculator.go [hunk 1]
```diff
@@ -175,7 +175,7 @@ func (c *ReplicaCalculator) GetMetricReplicas(currentReplicas int32, targetUtili
 }
 
 // calcPlainMetricReplicas calculates the desired replicas for plain (i.e. non-utilization percentage) metrics.
-func (c *ReplicaCalculator) calcPlainMetricReplicas(metrics metricsclient.PodMetricsInfo, currentReplicas int32, targetUtilization int64, namespace string, selector labels.Selector, resource v1.ResourceName) (replicaCount int32, utilization int64, err error) {
+func (c *ReplicaCalculator) calcPlainMetricReplicas(metrics metricsclient.PodMetricsInfo, currentReplicas int32, targetUsage int64, namespace string, selector labels.Selector, resource v1.ResourceName) (replicaCount int32, utilization int64, err error) {
```
Comment (pbeschetnov) [inline pkg/controller/podautoscaler/replica_calculator.go:178]: Also rename the variables passed to this method on lines 160, 173 (can't leave comments there for some reason), and up recursively as well.

Comment (kushagra98) [reply pkg/controller/podautoscaler/replica_calculator.go:178]: Addressed.

File: pkg/controller/podautoscaler/replica_calculator.go [hunk 2]
```diff
@@ -343,15 +343,15 @@ func (c *ReplicaCalculator) GetExternalMetricReplicas(currentReplicas int32, tar
...
+	usageRatio := float64(utilization) / float64(targetUsage)
 	replicaCount, timestamp, err = c.getUsageRatioReplicaCount(currentReplicas, usageRatio, namespace, podSelector)
 	return replicaCount, utilization, timestamp, err
 }
 
 // GetExternalPerPodMetricReplicas calculates the desired replica count based on a
 // target metric value per pod (as a milli-value) for the external metric in the
 // given namespace, and the current replica count.
-func (c *ReplicaCalculator) GetExternalPerPodMetricReplicas(statusReplicas int32, targetUtilizationPerPod int64, metricName, namespace string, metricSelector *metav1.LabelSelector) (replicaCount int32, utilization int64, timestamp time.Time, err error) {
+func (c *ReplicaCalculator) GetExternalPerPodMetricReplicas(statusReplicas int32, targetUsagePerPod int64, metricName, namespace string, metricSelector *metav1.LabelSelector) (replicaCount int32, utilization int64, timestamp time.Time, err error) {
```
Comment (pbeschetnov) [inline pkg/controller/podautoscaler/replica_calculator.go:354]: And this function returns `usagePerPod` (or `averageUsage`). Check please other functions you modified regarding this issue.

File: pkg/controller/podautoscaler/replica_calculator.go [hunk 3]
```diff
@@ -329,7 +329,7 @@ func (c *ReplicaCalculator) getReadyPodsCount(namespace string, selector labels.
 // GetExternalMetricReplicas calculates the desired replica count based on a
 // target metric value (as a milli-value) for the external metric in the given
 // namespace, and the current replica count.
-func (c *ReplicaCalculator) GetExternalMetricReplicas(currentReplicas int32, targetUtilization int64, metricName, namespace string, metricSelector *metav1.LabelSelector, podSelector labels.Selector) (replicaCount int32, utilization int64, timestamp time.Time, err error) {
+func (c *ReplicaCalculator) GetExternalMetricReplicas(currentReplicas int32, targetUsage int64, metricName, namespace string, metricSelector *metav1.LabelSelector, podSelector labels.Selector) (replicaCount int32, utilization int64, timestamp time.Time, err error) {
```
Comment (pbeschetnov) [inline pkg/controller/podautoscaler/replica_calculator.go:332]: I guess it now returns not the `utilization` but `usage`

File: pkg/controller/podautoscaler/replica_calculator.go [hunk 4]
```diff
@@ -251,13 +251,13 @@ func (c *ReplicaCalculator) calcPlainMetricReplicas(metrics metricsclient.PodMet
 
 // GetObjectMetricReplicas calculates the desired replica count based on a target metric utilization (as a milli-value)
 // for the given object in the given namespace, and the current replica count.
-func (c *ReplicaCalculator) GetObjectMetricReplicas(currentReplicas int32, targetUtilization int64, metricName string, namespace string, objectRef *autoscaling.CrossVersionObjectReference, selector labels.Selector, metricSelector labels.Selector) (replicaCount int32, utilization int64, timestamp time.Time, err error) {
+func (c *ReplicaCalculator) GetObjectMetricReplicas(currentReplicas int32, targetUsage int64, metricName string, namespace string, objectRef *autoscaling.CrossVersionObjectReference, selector labels.Selector, metricSelector labels.Selector) (replicaCount int32, utilization int64, timestamp time.Time, err error) {
```
Comment (pbeschetnov) [inline pkg/controller/podautoscaler/replica_calculator.go:254]: What about `GetObjectPerPodMetricReplicas` as well?

File: pkg/controller/podautoscaler/replica_calculator.go [hunk 5]
```diff
@@ -151,31 +151,31 @@ func (c *ReplicaCalculator) GetResourceReplicas(ctx context.Context, currentRepl
...
 	if err != nil {
 		return 0, 0, time.Time{}, fmt.Errorf("unable to get metrics for resource %s: %v", resource, err)
 	}
 
-	replicaCount, utilization, err = c.calcPlainMetricReplicas(metrics, currentReplicas, targetUtilization, namespace, selector, resource)
+	replicaCount, utilization, err = c.calcPlainMetricReplicas(metrics, currentReplicas, targetUsage, namespace, selector, resource)
 	return replicaCount, utilization, timestamp, err
 }
 
 // GetMetricReplicas calculates the desired replica count based on a target metric utilization
```
Comment (pbeschetnov) [inline pkg/controller/podautoscaler/replica_calculator.go:164]: utilization -> usage

File: pkg/controller/podautoscaler/replica_calculator.go [hunk 6]
```diff
@@ -151,31 +151,31 @@ func (c *ReplicaCalculator) GetResourceReplicas(ctx context.Context, currentRepl
 
 // GetRawResourceReplicas calculates the desired replica count based on a target resource utilization (as a raw milli-value)
```
Comment (pbeschetnov) [inline pkg/controller/podautoscaler/replica_calculator.go:152]: utilization -> usage

**Review by pbetkier — COMMENTED** [review]
I agree with comments from @pbeschetnov:

1) need to update method comments as well
2) need to update `utilization int64` return variables
3) also have a look at tests, perhaps the bad naming spilled over there to related tests as well

Basically anything that is called utilization and is of type int64 is suspicious.

File: pkg/controller/podautoscaler/replica_calculator.go (hunk 3 above)
Comment (kushagra98) [reply pkg/controller/podautoscaler/replica_calculator.go]: Addressed

File: pkg/controller/podautoscaler/replica_calculator.go (hunk 4 above)
Comment (kushagra98) [reply pkg/controller/podautoscaler/replica_calculator.go]: Addressed

File: pkg/controller/podautoscaler/replica_calculator.go (hunk 5 above)
Comment (kushagra98) [reply pkg/controller/podautoscaler/replica_calculator.go]: Addressed

**Comment by kushagra98** [comment]
Ohh. I just realised only one at a time can be a reviewer, if I click on other name it automatically removes another.

**Review by pbetkier — APPROVED** [review]
Now the code is much clearer, thanks :)

**Review by mwielgus — APPROVED** [review]
/lgtm
/approve
